### ISSUE TITLE
Add SF-NorMuon optimizer (schedule-free spectral)

### ISSUE THREAD

**Issue by mvillmow** [mvillmow/ml-odyssey#5649]
## Summary

Add **SF-NorMuon** to `odyssey.training.optimizers` — a modern optimizer
combining Schedule-Free anytime iterate averaging (three coupled y/z/x
sequences) with NorMuon's row-wise-normalized polar / Newton-Schulz update
geometry. The averaged iterate `x` is a good checkpoint at ANY step (no
learning-rate schedule), while the update direction is the row-normalized
orthogonalized momentum rather than the raw gradient.

This slots into the existing spectral / orthogonalization optimizer family
(`muon`, `normuon`, `mgup_muon`, `muon_hyperball`, `lionmuon`) and the
schedule-free family (`schedule_free`), reusing the shared
`newton_schulz_orthogonalize` helper.

## Update rule (per step t, 1-indexed)

```
y_t     = beta*x_t + (1-beta)*z_t                (schedule-free query point)
g_t     = grad(f, y_t)                            (caller supplies this)
m_t     = mu*m_{t-1} + (1-mu)*g_t                 (momentum EMA)
P_t     = rownorm( newton_schulz(m_t) )           (polar + per-row L2 norm)
z_{t+1} = (1 - lambda_wd)*z_t - eta*P_t           (WEIGHT DECAY ON z_t)
c_{t+1} = (r + 1)/(t + r + 1)                     (averaging weight)
x_{t+1} = (1 - c_{t+1})*x_t + c_{t+1}*z_{t+1}      (running average of z)
```

On step 1, initialize `z = x = params` (so `y_1 = params`) and
`m = zeros_like(params)`. `z`, `x`, `m` are caller-managed state; `y` is
recomputed each step and is where the caller must evaluate the gradient.

**Critical failure mode:** weight decay MUST apply to the fast iterate `z_t`,
never to the query point `y_t` or the average `x_t` — decaying `y_t` diverges
over long horizons (the schedule-free coupling re-injects the decayed magnitude
into the average). This is documented and covered by a dedicated test.

## Scope

- `src/odyssey/training/optimizers/sf_normuon.mojo` — `sf_normuon_step` (pure
  functional) + `sf_normuon_step_simple` convenience wrapper. Matrix-only
  (rank-2), reuses `newton_schulz_orthogonalize` from `muon.mojo`.
- `tests/odyssey/training/optimizers/test_sf_normuon.mojo` — guards +
  4-step parity to 1e-6 + weight-decay-on-z structural test.
- `tests/odyssey/training/optimizers/parity_refs/sf_normuon_parity_reference.py`
  — numpy reference generator (sibling convention).
- One-line/one-block wiring: optimizers `__init__` export, README primitive
  table row, `scripts/run_primitive_test.sh` registration.

## References

- Schedule-Free: Defazio, A., Yang, X. A., Mehta, H., Mishchenko, K., Khaled,
  A., & Cutkosky, A. (2024). *The Road Less Scheduled.* NeurIPS 2024.
  arXiv:2405.15682. <https://github.com/facebookresearch/schedule_free>
- NorMuon / Muon: Jordan et al. 2024, *Muon: An optimizer for hidden layers in
  neural networks.* <https://kellerjordan.github.io/posts/muon/>

The exact SF-NorMuon composition (schedule-free query point + row-normalized
polar update with weight decay on the fast iterate) is not described by a single
primary paper; the rule follows the specification pinned in the downstream
tracking issue.

Tracking: mvillmow/Random#79 [OPT-15]
